pdfbake: add -p

It was the one tool we run that could not open a password-protected PDF.
---
diff --git a/source/tools/pdfbake.c b/source/tools/pdfbake.c
index c255ebc61..ba6292698 100644
--- a/source/tools/pdfbake.c
+++ b/source/tools/pdfbake.c
@@ -38,6 +38,7 @@ static int usage(void)
 		"\t-A\tkeep annotations\n"
 		"\t-F\tkeep forms\n"
 		"\t-O -\tcomma separated list of output options\n"
+		"\t-p -\tpassword\n"
 	);
 	return 1;
 }
@@ -51,12 +52,13 @@ int pdfbake_main(int argc, char **argv)
 	int bake_widgets = 1;
 	char *output = "out.pdf";
 	char *flags = "garbage";
+	char *password = "";
 	char *input;
 	int c, code = EXIT_SUCCESS;
 
 	fz_optind = 1;
 
-	while ((c = fz_getopt(argc, argv, "AFO:")) != -1)
+	while ((c = fz_getopt(argc, argv, "AFO:p:")) != -1)
 	{
 		switch (c)
 		{
@@ -69,6 +71,9 @@ int pdfbake_main(int argc, char **argv)
 		case 'O':
 			flags = fz_optarg;
 			break;
+		case 'p':
+			password = fz_optarg;
+			break;
 		default:
 			return usage();
 		}
@@ -93,6 +98,9 @@ int pdfbake_main(int argc, char **argv)
 	fz_try(ctx)
 	{
 		doc = pdf_open_document(ctx, input);
+		if (pdf_needs_password(ctx, doc))
+			if (!pdf_authenticate_password(ctx, doc, password))
+				fz_throw(ctx, FZ_ERROR_ARGUMENT, "cannot authenticate password: %s", input);
 
 		pdf_bake_document(ctx, doc, bake_annots, bake_widgets);
 
